tests/main/bin-snap-in-snap/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: snaps bundling /bin/snap from snapd.deb can do so
+
+details: |
+  The snap binary may be embedded in a snap for as long as the bases match.
+  The snap binary may be called with LD_PRELOAD=, or other dynamic linker
+  environment variables without ill effects.
+
+  Note that this contract does *not* extend to the snap binary from the snapd
+  snap. That version cannot be embedded in snap applications.
+
+systems:
+  # This test runs only on systems that are ABI-matching the core24 snap.  This
+  # test could be extended to include more bases later. This test cannot run
+  # on ubuntu core, because /usr/bin/snap is the "snapd snap" version which is
+  # linked differently.
+  - ubuntu-24.04-64
+
+
+prepare: |
+  cp "$(command -v snap)" test-snapd-sh-core24/bin
```

Comment:
this could also copy the snap command the snapd snap and show that it fails to run

Reply:
Yes but then the test name would need to be different. I also don't know what that specific test would show. On some systems (without apparmor) it would actually work because it would load the dynamic linker from /snap/snapd/current

Reply:
This test is running only on ubuntu-24.04, which means that if you copy a snap binary from the snap it will fail (or at least should), which is the exact scenario folks should not be doing. i.e. copying stuff from the snapd snap into their snaps.